apps/prairielearn/src/tests/sync/questionsSync.test.ts

```diff
@@ -262,6 +274,230 @@ describe('Question syncing', () => {
+        a.email === newAuthor.email &&
+        a.orcid === newAuthor.orcid.replaceAll('-', ''),
+    );
+    assert.ok(author);
+
+    // Check that the question-author relationship was created
+    const questionAuthors = await util.dumpTableWithSchema(
+      'question_authors',
+      QuestionAuthorSchema,
+    );
+    const questionAuthor = questionAuthors.find(
+      (qa) => qa.question_id === originalSyncedQuestion.id && qa.author_id === author.id,
+    );
+    assert.ok(questionAuthor);
+  });
+
+  it('puts authors with origin course into database', async () => {
+    await features.enable('question-sharing');
+    const courseData = util.getCourseData();
+    const consumingCourseData = util.getCourseData();
+    consumingCourseData.course.name = 'CONSUMING 101';
+    consumingCourseData.questions = {}; // To prevent duplicate QIDs
+
+    await util.writeAndSyncCourseData(consumingCourseData);
+    await sqldb.execute(sql.set_sharing_name, {
+      course_name: consumingCourseData.course.name,
+      sharing_name: 'CONSUMING 101 SHARED',
+    });
+
+    const newAuthor = {
+      name: 'Example',
+      email: 'example@example.org',
+      orcid: '0000-0000-0000-0001',
+    };
+    const newAuthorWithOriginCourse = {
+      originCourse: 'CONSUMING 101 SHARED',
+    };
+
+    if (!courseData.questions[util.QUESTION_ID].authors) {
+      courseData.questions[util.QUESTION_ID].authors = [];
+    }
+    courseData.questions[util.QUESTION_ID].authors.push(newAuthor, newAuthorWithOriginCourse);
+
+    const courseDir = await util.writeCourseToTempDirectory(courseData);
+    await util.syncCourseData(courseDir);
+
+    const originalSyncedQuestion = await findSyncedQuestion(util.QUESTION_ID);
+    assert.ok(originalSyncedQuestion);
+
+    // Check that the author was added to the authors table
+    const authors = await util.dumpTableWithSchema('authors', AuthorSchema);
+    const author1 = authors.find(
+      (a) =>
+        a.author_name === newAuthor.name &&
+        a.email === newAuthor.email &&
+        a.orcid === newAuthor.orcid.replaceAll('-', ''),
+    );
+    const author2 = authors.find((a) => a.origin_course === '1'); // Any easy way to avoid hardcoding the number?
+    assert.ok(author1);
+    assert.ok(author2);
+    assert.notDeepEqual(author1, author2);
+
+    // Check that the question-author relationship was created
+    const questionAuthors = await util.dumpTableWithSchema(
+      'question_authors',
+      QuestionAuthorSchema,
+    );
+
+    const questionAuthor1 = questionAuthors.find(
+      (qa) => qa.question_id === originalSyncedQuestion.id && qa.author_id === author1.id,
+    );
+    const questionAuthor2 = questionAuthors.find(
+      (qa) => qa.question_id === originalSyncedQuestion.id && qa.author_id === author2.id,
+    );
+    assert.ok(questionAuthor1);
+    assert.ok(questionAuthor2);
+  });
+
+  it('records a warning if "authors" object is invalid', async () => {
```

Comment:
The test name mentions "warning" but you're actually making an assertion about the presence of errors.